lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -193,11 +584,12 @@ export default function getAPIImplementation(): APIInterface {
             }
             let user = response.user;
 
+            let recipeUser = user.loginMethods.find((u) => u.recipeId === "emailpassword" && u.email === email);
             let session = await Session.createNewSession(
                 options.req,
                 options.res,
                 user.id,
-                user.recipeUserId,
+                recipeUser?.recipeUserId,
```

Comment:
recipeUser should never be undefined here right? So instead of doing `?`, we should check if it's undefined and if it is, throw an error like should never come here